src/auth/services/users.service.ts

```diff
@@ -144,4 +162,112 @@ export class UserService {
       }),
     );
   }
+
+  async findOthersProfiles(profileFilter: GetProfilesFilterDto) {
+    const { search } = profileFilter;
+
+    const query = this.userRepository.createQueryBuilder('user');
+
+    if (search) {
+      query.where(
+        '(LOWER(user.firstname) LIKE LOWER(:search) OR LOWER(user.lastname) LIKE LOWER(:search))',
+        {
+          search: `%${search}%`,
+        },
+      );
+    }
+
+    const profiles = await query.getMany();
+    return profiles;
+  }
+
+  async updateProfile(
+    user: User,
+    updateProfileDto: UpdateProfileDto,
+  ): Promise<User> {
+    const profile = await this.userRepository.findOne({
+      where: [{ id: user.id }],
+    });
+
+    if (!profile) {
+      throw new NotFoundException('Not Found');
+    } else {
+      const { firstname, lastname, gender } = updateProfileDto;
+
+      profile.firstname = firstname;
+      profile.lastname = lastname;
+      profile.gender = gender;
+
+      await this.userRepository.save(profile);
```

Comment:
Sao k xài thê snayf luôn e?
 await this.userRepository.save(updateProfileDto); ?

Reply:
Dạ nếu vậy thì nó bị lỗi là `Field 'email' doesn't have a default value` ạ.